tests/StreamTest.php

```diff
@@ -158,4 +158,34 @@ public function testDoesNotThrowInToString()
         $s = new NoSeekStream($s);
         $this->assertEquals('foo', (string) $s);
     }
+
+    public function testStreamReadingWithZeroLength()
+    {
+        $r = fopen('php://temp', 'r');
+        $stream = new Stream($r);
+
+        $this->assertEquals('', $stream->read(0));
+
+        $stream->close();
+    }
+
+    public function testStreamReadingWithNegativeLength()
```

Comment:
Please just add 

```
@expectedException \RuntimeException
@expectedExceptionMessage Length parameter must not be negative
```

This will not properly close the stream explicitly. But we can just add try...finally when raising requirement to PHP >= 5.5